content/asp.net/getting-started/files.md

```diff
@@ -3,13 +3,76 @@ by [Steve Smith](http://deviq.com/me/steve-smith)
+    {
+        public static void Main(string[] args)
+        {
+            var host = new WebHostBuilder()
+                .UseKestrel()
+                .UseStartup<Startup>()
+                .Build();
+
+            host.Run();
+        }
+    }
+}
+```
+
+Note that the above class references the ``Startup`` type, which is where you configure your individual app.
+
+## Startup.cs
+
+The *Startup.cs* file contains the application's ``Startup`` class. This class must include certain methods that will be called by the host (that you just saw was configured in *Program.cs*). Specifically, the class must define a ``Configure`` method. It can optionally include a ``ConfigureServices`` method and/or a constructor as well.
+
+### Configure
+
+The ``Configure`` method is used to configure how your app will handle requests. This is sometimes referred to as the app's *request pipeline*, and involves configuring one or more pieces of *middleware*. At its most basic, an ASP.NET Core app can configure a single handler that will respond to every request, like this one:
+
+```c#
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+
+namespace ConsoleApplication
+{
+    public class Startup
+    {
+        public void Configure(IApplicationBuilder app)
+        {
+            app.Run(context =>
+            {
+                return context.Response.WriteAsync("Hello from ASP.NET Core!");
+            });
+        }
+    }
+}
+```
+
+The ``app.Run`` method configures middleware, which you'll learn more about in the next lesson. In addition to configuring your app's request pipeline, the ``Configure`` method is also used to (surprise!) configure other aspects of the app. It's here that you'll add support for diagnostics, logging, error handling, routing, and frameworks like MVC. These packages often have helper methods that follow a naming pattern of ``Use[Package]``, which is used to configure them within the ``Configure`` method. For example, configuring MVC is done with the ``app.UseMvc();`` statement. You'll see examples of these packages in later lessons in this tutorial. 
+
+### ConfigureServices
+
+The ``ConfigureServices`` method in the ``Startup`` class is optional. You use it to populate the services *container* used by ASP.NET Core. This container provides requested services to objects within your app through a process called [dependency injection](http://deviq.com/dependency-injection/). The container is represented by the ``IServiceCollection`` parameter that is passed into the ``ConfigureServices`` method. ASP.NET Core provides and requests many services itself, and uses the same infrastructure you will use in your app. Many built-in as well as third-party features require certain services to work properly, and you'll often see a particular package (for instance, MVC) adding services in ``ConfigureServices`` and also being configured in ``Configure``.
```

Comment:
>Many built-in as well as third-party features require certain services to work properly

Suggestion: Many built-in as well as third-party features require the addition and sometimes configuration of certain services to work properly